pkg/cnbutils/auth_test.go

```diff
@@ -18,41 +18,61 @@ func TestGenerateCnbAuth(t *testing.T) {
 	}
 
 	t.Run("successfully generates cnb auth env variable", func(t *testing.T) {
-		mockUtils.AddFile("/test/valid_config.json", []byte("{\"auths\":{\"example.com\":{\"auth\":\"dXNlcm5hbWU6cGFzc3dvcmQ=\"}}}"))
-		auth, err := cnbutils.GenerateCnbAuth("/test/valid_config.json", mockUtils)
+		mockUtils.AddFile("/test/valid_config.json", []byte("{\"auths\":{\"https://example.com/\":{\"auth\":\"dXNlcm5hbWU6cGFzc3dvcmQ=\"}}}"))
+		auth, err := cnbutils.ParseDockerConfig("/test/valid_config.json", mockUtils)
 		assert.NoError(t, err)
-		assert.Equal(t, "{\"example.com\":\"Basic dXNlcm5hbWU6cGFzc3dvcmQ=\"}", auth)
+
+		authString, err := auth.ToCNBString()
+		assert.NoError(t, err)
+		assert.Equal(t, "{\"example.com\":\"Basic dXNlcm5hbWU6cGFzc3dvcmQ=\"}", authString)
+		assert.True(t, auth.AuthExistsForImage("example.com/foo/bar:123"))
+		assert.False(t, auth.AuthExistsForImage("docker.io/foo/bar"))
 	})
 
 	t.Run("successfully generates cnb auth env variable from username and password", func(t *testing.T) {
 		mockUtils.AddFile("/test/valid_config.json", []byte("{\"auths\":{\"example.com\":{\"username\":\"username\",\"password\":\"password\"}}}"))
-		auth, err := cnbutils.GenerateCnbAuth("/test/valid_config.json", mockUtils)
+		auth, err := cnbutils.ParseDockerConfig("/test/valid_config.json", mockUtils)
+		assert.NoError(t, err)
+
+		authString, err := auth.ToCNBString()
 		assert.NoError(t, err)
-		assert.Equal(t, "{\"example.com\":\"Basic dXNlcm5hbWU6cGFzc3dvcmQ=\"}", auth)
+		assert.Equal(t, "{\"example.com\":\"Basic dXNlcm5hbWU6cGFzc3dvcmQ=\"}", authString)
+		assert.True(t, auth.AuthExistsForImage("example.com/foo/bar:123"))
```

Comment:
```suggestion
		assert.True(t, auth.AuthExistsForImage("example.com/foo/bar:123"))
		assert.True(t, auth.AuthExistsForImage("registry.example.com/foo/bar:123"))
```

Reply:
the result here will be `false`, since keychain's `Resolve` function expects full match